internal/trie/node/hash.go

```diff
@@ -5,51 +5,93 @@ package node
 
 import (
 	"bytes"
+	"fmt"
 
 	"github.com/ChainSafe/gossamer/internal/trie/pools"
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
-// EncodeAndHash returns the encoding of the node and
-// the blake2b hash digest of the encoding of the node.
-// If the encoding is less than 32 bytes, the hash returned
-// is the encoding and not the hash of the encoding.
-func (n *Node) EncodeAndHash(isRoot bool) (encoding, hash []byte, err error) {
-	if !n.Dirty && n.Encoding != nil && n.HashDigest != nil {
-		return n.Encoding, n.HashDigest, nil
+// MerkleValue produces the Merkle value from the encoding of a non-root node.
+// If the encoding is less or equal to 32 bytes, the Merkle value is the encoding.
+// Otherwise, the Merkle value is the Blake2b hash digest of the encoding.
+func MerkleValue(encoding []byte) (merkleValue []byte, err error) {
```

Comment:
any specific reason why we are exporting this?

Reply:
sorry I forgot to mention, it's needed in the `lib/trie/proof` code on #2604

Reply:
This now used in `lib/trie/proof` within this PR since #2604 got merged.